### DISCUSSION TITLE
reCaptcha - "Failed to validate captcha" error

### DISCUSSION THREAD

**Discussion by Garpo001** [OrchardCMS/OrchardCore#10159] [Q&A]
I have setup reCaptcha on OrchardCore v1.0.0. with v2 keys but keep getting "Failed to validate captcha" error. Is there something more I have to do like setup workflow etc. This is standard user functionality i.e. Forgot password, Register etc.

![image](https://user-images.githubusercontent.com/4099702/130446781-3d3eb09a-d0fc-422b-b3c7-a48e64c6ebc7.png)

**Comment by hishamco** [comment]
I think it's duplicate for #9857 #9519

**Reply by hishamco** [reply]
@Skrypt we need to close two and keep one for tracking. Am I right?

**Reply by Skrypt** [reply]
Keep it open. I can't repro this issue at all. Need to repro first. This can be related to the domain not configured in the ReCaptcha settings on Google website else right now I don't know where the issue is here as I can't repro and it works fine on my side.

**Reply by Garpo001** [reply]
@Skrypt I am using a Google Enterprise account. Could this make a difference in the way a captcha is authenticated?

**Reply by Garpo001** [reply]
Ok, I confirmed that using Enterprise even when creating V2 key's does not validate. However using normal v2 Checkbox key's works out of the box.

**Reply by Skrypt** [reply]
Any documentation that I should be looking at? What's the example of javascript to implement in the Google Entreprise Account for a Recaptcha?

**Reply by Garpo001** [reply]
https://cloud.google.com/recaptcha-enterprise/docs

**Reply by Skrypt** [reply]
Yeah looking at the doc it is more likely it will never work with the current module. This would require to be implemented. And the code samples shows server side code but nothing about the current javascript widget that is added on the HTML side. Maybe I would need to create an account for that.

**Reply by Garpo001** [reply]
Agreed. Not sure how much work has been done around the v3 implementation yet, but we will be able to look at spending some time assisting with the implementation. Not promising, but maybe we can collaborate and at the same time see if we can help with the Enterprise implementation also and submit a PR.

**Reply by Garpo001** [reply]
It seems Google split out the api endpoints and added an additional "enterprise" part to the path. Maybe we can add a version indicator on the Admin Config to qualify between Legacy and Enterprise  in addition to the key's to qualify which version and then render endpoints accordingly. Enterprise can also still distinguish between checkbox type and the new unobtrusive version so maybe have a qualifier for that as well to distinguish for which keys were generated.
